### PR TITLE
fixed JSB_glGetUniformfv uniform location is different from index.

### PR DESCRIPTION
**Pull request by fusijie**

### CODE REVIEW

File: cocos/scripting/js-bindings/manual/jsb_opengl_manual.cpp [hunk 1]
```diff
@@ -431,13 +431,20 @@ bool JSB_glGetUniformfv(JSContext *cx, uint32_t argc, jsval *vp)
...
     GLchar* namebuffer = new GLchar[length+1];
     GLint size = -1;
     GLenum type = -1;
 
-    glGetActiveUniform(arg0, arg1, length, NULL, &size, &type, namebuffer);
+    for(int i = 0; i  <  activeUniforms; ++i)
+    {
+        glGetActiveUniform(arg0, i, length, NULL, &size, &type, namebuffer);
+        if(arg1 == glGetUniformLocation(arg0, namebuffer)) break;
+    }
```
Comment (dabingnn) [inline cocos/scripting/js-bindings/manual/jsb_opengl_manual.cpp:457]: The code is unsafe.
It is better to deal with invalid location.

**Comment by dabingnn** [comment]
Pull request reviewed, it is ready to be merged.

File: cocos/scripting/js-bindings/manual/jsb_opengl_manual.cpp (hunk 1 above)
Comment (fusijie) [inline cocos/scripting/js-bindings/manual/jsb_opengl_manual.cpp:457]: Thanks. I have fixed it.

**Comment by dabingnn** [comment]
Pull request reviewed, it is ready to be merged.

**Pull request merged**